includes/class-wc-taxjar-nexus.php

```diff
@@ -142,4 +143,25 @@ public function get_nexus_from_cache() {
 		return $response;
 	}
 
+	/**
+	 * Clear non US nexus states from rates table to prevent tax calculation when nexus is removed in a state
+	 * @param $regions - array of nexus regions from API request
+	 */
+	public function clear_non_nexus_rates( $regions ) {
+		global $wpdb;
+		$nexus_states = array();
+
+		foreach( $regions as $region ) {
+			if ( ! empty( $region->country_code ) && $region->country_code === 'US' ){
+				if ( ! empty( $region->region_code ) ) {
+					$nexus_states[] = $region->region_code;
+				}
+			}
+		}
+
+		$nexus_states_string = join( "','", $nexus_states );
+		$query = "DELETE FROM {$wpdb->prefix}woocommerce_tax_rates WHERE tax_rate_country = 'US' AND tax_rate_state NOT IN ('{$nexus_states_string}')";
```

Comment:
So we purge tax rates after installing and back them up. Have we seen or anticipated a scenario where a customer may want to enter a manual tax rate for a non-nexus US state after installation, such as using a flat rate instead of having TaxJar calculate? We could update the WHERE clause to check for `%s Tax` format in the `tax_rate_name`, but it may be overcautious.

Reply:
I haven't seen or heard of that scenario before, though it could be possible. I feel fine with the code as is right now because I can't think of a scenario where someone would want to take that approach. Plus I think we want to discourage people from altering the rates in that way for a US location. They are basically ensuring they will not have the most accurate rates.

Reply:
@dallendalton Sounds good, resolving this 👍